validation/util/test.go

```diff
@@ -21,6 +22,47 @@ var (
 	RuntimeCommand = "runc"
 )
 
+// LifecycleAction defines the phases will be called.
+type LifecycleAction int
+
+const (
+	// LifecycleActionCreate creates a container
+	LifecycleActionCreate = 1 << iota
+	// LifecycleActionStart starts a container
+	LifecycleActionStart
+	// LifecycleActionDelete deletes a container
+	LifecycleActionDelete
+)
+
+// LifecycleStatus follows https://github.com/opencontainers/runtime-spec/blob/master/runtime.md#state
+type LifecycleStatus int
+
+const (
+	// LifecycleStatusCreating "creating"
+	LifecycleStatusCreating = 1 << iota
+	// LifecycleStatusCreated "created"
+	LifecycleStatusCreated
+	// LifecycleStatusRunning "running"
+	LifecycleStatusRunning
+	// LifecycleStatusStopped "stopped"
+	LifecycleStatusStopped
+)
+
+var lifecycleStatusMap = map[string]LifecycleStatus{
+	"creating": LifecycleStatusCreating,
```

Comment:
Or these could just be strings:

```go
type LivecycleStatus string

const (
  LifecycleStatusCreating = "creating"
  …
)
```

and then you cast with `LifecycleStatus(state.Status)`.  I don't see a need for indirection through an integer.

Reply:
An 'integer' is useful when we expect to have multiple statuses. For example, if we want to delete a container, we expected to have  'created' or 'stopped' status. So we can 'WaitingForStatus(Created|Stopped)'.